File: MS.NET/Fundamentals/Runtime/Reflection/interval.cs
using System;

partial class Interval
{
	private int min, sec;

	partial void OnInitialize();

	public Interval(int m, int s)
	{
		min = m + s / 60;
		sec = s % 60;
		OnInitialize();
	}

	public int Minutes => min;

	public int Seconds => sec;

	public int GetTime() => 60 * min + sec;
	
	public override string ToString()
	{
		return $"{min}:{sec:00}"; //string.Format("{0}:{1:00}", min, sec)
	}

	public override int GetHashCode()
	{
		return 1000 * GetTime();
	}

	public override bool Equals(object other)
	{
		if(other is Interval that)
			return (this.min == that.min) && (this.sec == that.sec);
		
		return false;
	}
		
}

File: Java/Fundamentals/Runtime/Reflection/Interval.java
class Interval{

	private int minute, second;

	public Interval(int m, int s){
		minute = m + s / 60;
		second = s % 60;
	}

	public int minutes(){
		return minute;
	}

	public int seconds(){
		return second;
	}

	public int time(){
		return 60 * minute + second;
	}

	public String toString(){
		return String.format("%d:%02d", minute, second);
	}

	public int hashCode(){
		return 1000 * time();
	}

	public boolean equals(Object other){
		if(other instanceof Interval){
			Interval that = (Interval)other;
			return (this.minute == that.minute) && (this.second == that.second);
		}
		return false;
	}
}


